### PR TITLE
New Cheat Sheet: C Statements

### PR DESCRIPTION
**Pull request by sriranganathan**
## Type of Change
- [x] New Instant Answer
  - [X] Cheat Sheet
- [ ] Improvement
  - [ ] Bug fix
  - [ ] Enhancement
- [ ] Non–Instant Answer
  - [ ] Other (Role, Template, Test, Documentation, etc.)
## Description of new Instant Answer, or changes

Add a new cheat sheet for C Statements like while, do-while, for, if, switch, etc.
## People to notify

@moollaza 

---

Forum Topic: https://forum.duckduckhack.com/t/new-c-statements-cheat-sheet/1092
IA Page: https://duck.co/ia/view/c_statements_cheat_sheet

### CODE REVIEW

**Comment by daxtheduck** [comment]
@sriranganathan Thanks for the pull request! Unfortunately, we're only accepting contributions and improvements to a limited portion of the search space.
            
The staff and community are currently focused on making DuckDuckGo the [best search engine for programmers](https://forum.duckduckhack.com/t/duckduckhack-programming-mission-overview/53). To ensure we're making as much progress as possible, **we are only accepting Instant Answers and updates related to the programming mission**.
    
If this is not related to the Programming Mission, it will likely be closed but you can:
    
1. Help us prevent further confusion by telling us where you found out about DuckDuckHack or where we could've alerted you earlier.
2. Keep this code handy! We'll be expanding into other search topics in the future!
3. Join us at our new home: https://forum.duckduckhack.com/

If this is related to the Programming Mission, it will be reviewed by staff and the community as soon as possible!

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?u8m27EOlpZ)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?PWp7TGTtsh)

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        ],
+        "Iteration Statements": [
+            {
+                "key": "while ( expression ) statement",
+                "val": "while Loop"
+            },
+            {
+                "key": "do statement while ( expression ) ;",
+                "val": "do-while Loop"
+            },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: Similarly here, `Executes a statement repeatedly until the value of condition becomes false; The test takes place after each iteration`

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "for ( init_clause ; expression(optional) ; expression(optional) ) statement",
+                "val": "for Loop"
+            }
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: Also here, `Executes a loop until the expression becomes false; used as a shorter equivalent of while loop`

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        ],
+        "Jump Statements": [
+            {
+                "key": "break ;",
+                "val": "break Statement"
+            },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: `Causes the enclosing for, while or do-while loop or switch statement to terminate`

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "continue ;",
+                "val": "continue Statement"
+            },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: `Causes the remaining portion of the enclosing for, while or do-while loop body to be skipped; it causes a jump to the end of the loop body`

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "return expression(optional) ;",
+                "val": "return Statement with an optional expression"
+            },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: `Terminates the current function and returns specified (optional) value to the caller function`

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "goto identifier ;",
+                "val": "goto Statement"
+            }
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: `Transfers control unconditionally to the desired location, identified by the label`

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?LkFF3cn0ll)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?ethJfUqHNe)

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?Qj0ghsigFi)

**Comment by sriranganathan** [comment]
@rasikapohankar Added the descriptions

**Comment by mintsoft** [comment]
@sriranganathan Originally on the forum this was suggested as a fathead rather than a cheatsheet; I'm not sure if @moollaza has different thoughts on this now?

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 7]
```diff
@@ -49,33 +49,33 @@
         "Iteration Statements": [
             {
                 "key": "while ( expression ) statement",
-                "val": "while Loop"
+                "val": "while Loop - Executes a statement repeatedly, until the value of expression becomes false; The test takes place before each iteration"
             },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: @sriranganathan the `while Loop - ` part can be removed from the val here. Similarly please remove from the other statements as well

File: share/goodie/cheat_sheets/json/c-statements.json [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+            {
+                "key": "expression(optional) ;",
+                "val": "An expression followed by a semicolon. An expression statement without an expression is called a null statement"
+            }
+        ],
+        "Selection Statements": [
+            {
+                "key": "if ( expression ) statement",
+                "val": "If statement"
+            },
```
Comment (rasikapohankar) [inline share/goodie/cheat_sheets/json/c-statements.json]: Lets add descriptions about what the statement is similar to the iteration statements. `Conditionally executes code; used where code needs to be executed only if some condition is true`
Can you also add such descriptions for the other selection statements and `Labels` section?

Comment (sriranganathan) [inline share/goodie/cheat_sheets/json/c-statements.json]: @rasikapohankar Done

File: share/goodie/cheat_sheets/json/c-statements.json (hunk 7 above)
Comment (sriranganathan) [inline share/goodie/cheat_sheets/json/c-statements.json]: @rasikapohankar Done

**Comment by daxtheduck** [comment]
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=C Statements Cheat Sheet)
                
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/c_statements_cheat_sheet_mobile.png?0zZT9SdyUL)

**Comment by daxtheduck** [comment]
@sriranganathan this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

**Comment by gautamkrishnar** [comment]
@sriranganathan 👍 Merged... Thanks for your contribution...

**Pull request merged**